src/client.ts

```diff
@@ -587,4 +589,15 @@ export default class Client implements ClientInterface {
   private hasUserDefinedCustomBundle(): boolean {
     return this.customBundleGemfile.length > 0;
   }
+
+  private diagnosticPullOptions(): DiagnosticPullOptions {
+    const configuration = vscode.workspace.getConfiguration("rubyLsp");
+    const pullOn: "change" | "save" | "both" =
```

Comment:
Is there are a chance another option could be added in future, in which case `both` would no longer make sense? If so, could this be instead be named `change_or_save`, or similar?

Reply:
If you open a new file, diagnostics are always computed once in the beginning. Until the LSP clears the diagnostics for that file, they never go away - even if you switch files.

I think we can go with `both` and change in the future if it ever becomes necessary.